xilem/src/view/checkbox.rs

```diff
@@ -26,28 +29,29 @@ pub struct Checkbox<F> {
     callback: F,
 }
 
-impl<F, State, Action> MasonryView<State, Action> for Checkbox<F>
+impl<F, State, Action> View<State, Action, ViewCtx> for Checkbox<F>
 where
     F: Fn(&mut State, bool) -> Action + Send + Sync + 'static,
 {
-    type Element = masonry::widget::Checkbox;
+    type Element = Pod<widget::Checkbox>;
     type ViewState = ();
 
-    fn build(&self, cx: &mut ViewCtx) -> (WidgetPod<Self::Element>, Self::ViewState) {
+    fn build(&self, cx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
         cx.with_leaf_action_widget(|_| {
             WidgetPod::new(masonry::widget::Checkbox::new(
                 self.checked,
                 self.label.clone(),
             ))
+            .into()
         })
     }
 
     fn rebuild(
         &self,
-        _view_state: &mut Self::ViewState,
-        cx: &mut ViewCtx,
         prev: &Self,
-        mut element: WidgetMut<Self::Element>,
+        (): &mut Self::ViewState,
+        cx: &mut ViewCtx,
+        mut element: WidgetMut<'_, widget::Checkbox>,
```

Comment:
Not sure whether we should generally use the qualified `View::Element::Mut` path or the concrete type, I used latter, because I think it's slightly more readable.

Reply:
Agreed, the "resolved" version is better.

The reason I didn't before is only because rust-analyzer autofills it to the referential version